jax/_src/sharding.py

```diff
@@ -49,8 +49,7 @@ def device_set(self) -> Set[Device]:
     raise NotImplementedError('Subclasses should implement this method.')
 
   @abc.abstractmethod
-  def devices_indices_map(
-      self, global_shape: Shape) -> Mapping[Device, Optional[Index]]:
+  def devices_indices_map(self, global_shape: Shape) -> Mapping[Device, Index]:
```

Comment:
I think this should stay as a `Mapping[Device, Optional[Index]]`; it's part of the API. Was there an issue with that?

Reply:
Every concrete implementation of this method is annotated `Mapping[Device, Index]` (https://cs.github.com/google/jax?q=%22def+devices_indices_map%22), and there are several places in the code where the `None` possibility is unhandled, and these are flagged by mypy under the old implementation.

Reply:
Switched back after chatting with @yashk2810